### ISSUE TITLE
`linuxdeployqt`  doesn't always include all QtQuick files

### ISSUE THREAD

**Issue by kubark42** [probonopd/linuxdeployqt#102]
`MyAwesomeQtApp` depends on both QtQuick and QtQuick2.0. It runs fine after a normal build.
Strangely, if I am in my `build` directory, which is one level above the source code, and I run 

```
linuxdeployqt-continuous-x86_64.AppImage MyAwesomeQtApp -qmldir=../
```

And it all works. Yay!

However,

```
linuxdeployqt-continuous-x86_64.AppImage MyAwesomeQtApp -qmldir=../ -appimage
```

or 

```
linuxdeployqt-continuous-x86_64.AppImage MyAwesomeQtApp -qmldir=../ -bundle-non-qt-libs
```

leads to 

```
./AppRun 
QQmlApplicationEngine failed to load component
qrc:/parc-ul.qml:37 Type MissionForm unavailable
qrc:/qml/MissionForm.qml:100 Type CCalendar unavailable
qrc:/qml/CCalendar.qml:2 module "QtQuick.Controls" version 1.4 is not installed
```

If I look in the directory of the successful `linuxdeployqt` run, then I see that there are files in `
`~/Documents/sandbox/PARC-ul/build/qml/QtQuick/Controls`. However, in the "failed" runs, there are no files in that directory. But manually copying over the files, with `cp ../build.works/qml/QtQuick/Controls/* ./qml/QtQuick/Controls/`, and everything works.

P.S. Unrelated, but I have the same problem as identified in https://github.com/probonopd/linuxdeployqt/issues/25.

**Comment by kubark42** [comment]
So weird. Now `linuxdeployqt-continuous-x86_64.AppImage MyAwesomeQtApp -qmldir=../` also doesn't work properly anymore. There's some combination of commands which yielded the working results, though, grr...

**Comment by kubark42** [comment]
Further update. I can get the AppImage to work by copying in the missing files before running `linuxdeployqt-continuous-x86_64.AppImage MyAwesomeQtApp -qmldir=../ -bundle-non-qt-libs -appimage` a second time.

**Comment by probonopd** [comment]
Please try with an absolute path instead of `-qmldir=../` - does it make any difference?

**Comment by probonopd** [comment]
@kubark42 is your application available for testing?

**Comment by kubark42** [comment]
> Please try with an absolute path instead of -qmldir=../ - does it make any difference?

Lemme give that a shot. It's on a testing server so it might take me a bit to try that approach.

> is your application available for testing?
It's an internal project, so not available for external testing. :(

**Comment by kubark42** [comment]
Yes, identical behavior when using an absolute path. The other QML files are found and copied, just not the ones in QtQuick 1.0

**Comment by andeplane** [comment]
I have the same problem in my project (which is available [github.com/ovilab/atomify](github.com/ovilab/atomify)).

**Comment by probonopd** [comment]
Run the exact same command twice, does it make a difference? Sometimes we need 2 runs.

**Comment by andeplane** [comment]
Nope, that didn't solve this problem, but was my solution to another problem actually (if I run the command once, I get runtime problems with OpenGL context). But the Quick Controls 1.4 problem is still here. My solution is to do `cp /path/to/qt/qml/QtQuick/Controls/* qml/QtQuick/Controls/`, so I'm happy enough. But as I said, this is also a problem with macdeployqt. I have planned to dig into it, but haven't thus far.

I must say thanks though, because this tool really helps! 

If you're interested, my project linked to above (rather big though) reproduces the problem on my machine. Build instructions in the readme.

**Comment by probonopd** [comment]
What did you specify as `-qmldir=`? Try experimenting with that. You can specify multiple ones.

**Comment by andeplane** [comment]
That might solve the problem, but main.qml contains both 2.2 and 1.4 so ideally the tool should fix this. The deployment script I have that works can be found [here](https://github.com/ovilab/atomify/blob/dev/deploy/deploy_linux.sh). I have also tried a full path to the qml files.
